crates/nix_health/src/check/direnv.rs

```diff
@@ -93,71 +133,99 @@ fn activation_check(local_flake: &std::path::Path, required: bool) -> Check {
-        let mut bash_path = None;
-        let mut direnv_config = None;
-        // NOTE: One day we'll switch to using JSON output
-        // https://github.com/direnv/direnv/pull/1142
-        for line in out.lines() {
-            if let Some(path) = line.strip_prefix("bash_path ") {
-                bash_path = Some(PathBuf::from(path));
-            }
-            if let Some(config_dir) = line.strip_prefix("DIRENV_CONFIG ") {
-                let config_file = PathBuf::from(config_dir).join("direnvrc");
-                // Read config_file and assign to direnv_config
-                if config_file.exists() {
-                    let config = std::fs::read_to_string(config_file)?;
-                    direnv_config = Some(config);
-                }
-            }
-        }
-        Ok(Self {
-            bin_path,
-            direnv_config,
-            bash_path,
-        })
+        let status = DirenvStatus::from_json(&out)?;
+        Ok(status.state.is_found_rc_allowed())
+    } else {
+        anyhow::bail!("Unable to run direnv status --json: {:?}", output.stderr)
     }
 }
 
-/// Check if direnv was already activated in [project_dir]
-
-pub fn is_direnv_active_on(project_dir: &std::path::Path) -> anyhow::Result<bool> {
+fn direnv_version() -> anyhow::Result<String> {
     let output = std::process::Command::new("direnv")
-        .arg("status")
-        .current_dir(project_dir)
+        .args(["--version"])
         .output()?;
     if output.status.success() {
         let out = String::from_utf8_lossy(&output.stdout);
-        let mut allowed = false;
-        for line in out.lines() {
-            if line == "Found RC allowed true" {
-                allowed = true;
-                break;
-            }
-        }
-        Ok(allowed)
+        let trimmed_out = out.trim();
+        Ok(trimmed_out.to_string())
     } else {
-        anyhow::bail!("Unable to run direnv status: {:?}", output.stderr)
+        anyhow::bail!("Unable to run direnv --version: {:?}", output.stderr)
+    }
+}
+
+/// Information about the direnv status
+#[derive(Debug, Serialize, Deserialize, Clone)]
+struct DirenvStatus {
+    config: DirenvConfig,
+    state: DirenvState,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+struct DirenvConfig {
+    /// Path to the config folder of direnv
+    #[serde(rename = "ConfigDir")]
+    config_dir: PathBuf,
+    /// Path to the direnv binary
+    #[serde(rename = "SelfPath")]
+    self_path: PathBuf,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+struct DirenvState {
+    /// Information about the .envrc found in the current directory
+    #[serde(rename = "foundRC")]
+    found_rc: Option<DirenvRC>,
+    /// Information about the .envrc that is currently allowed using `direnv allow`
+    #[serde(rename = "loadedRC")]
+    loaded_rc: Option<DirenvRC>,
+}
+
+impl DirenvState {
+    /// Check if the .envrc file is allowed
+    fn is_found_rc_allowed(&self) -> bool {
```

Comment:
```suggestion
    fn is_allowed(&self) -> bool {
```

Why? Because function name should reflect semantics not implementation.